### DISCUSSION TITLE
Can't sign out properly

### DISCUSSION THREAD

**Discussion by MrSpock** [kanidm/kanidm#2760] [Q&A]
Hi,
I'm evaluating kanidm and I stuck with such basic stuff like logout.
When I login to system and click Sign Out I'm getting following log error and web message:
`kanidmd  | 76071ae1-1f1b-40b6-a9f8-d1d2778420b0 INFO     request [ 4.50ms | 9.87% / 100.00% ] method: GET | uri: /v1/logout | version: HTTP/1.1
kanidmd  | 76071ae1-1f1b-40b6-a9f8-d1d2778420b0 INFO     ┝━ handle_logout [ 4.06ms | 22.44% / 90.13% ]
kanidmd  | 76071ae1-1f1b-40b6-a9f8-d1d2778420b0 INFO     │  ┝━ validate_client_auth_info_to_ident [ 3.05ms | 67.69% ]
kanidmd  | 76071ae1-1f1b-40b6-a9f8-d1d2778420b0 INFO     │  │  ┕━ ｉ [info]: A valid limited session value exists for this token | event_tag_id: 10
kanidmd  | 76071ae1-1f1b-40b6-a9f8-d1d2778420b0 INFO     │  ┝━ ｉ [info]: modify initiator | event_tag_id: 10 | name: User( spock@*, fc4a6f61-a843-4dbf-bf27-cfc91b0ec0f6 ) (d8ebf915-0b57-4439-b3d9-ea757cf7585e, read write)
kanidmd  | 76071ae1-1f1b-40b6-a9f8-d1d2778420b0 INFO     │  ┝━ ｉ [info]: search | event_tag_id: 10 | initiator: User( spock@idm*, fc4a6f61-a843-4dbf-bf27-cfc91b0ec0f6 ) (d8ebf915-0b57-4439-b3d9-ea757cf7585e, read write)
kanidmd  | 76071ae1-1f1b-40b6-a9f8-d1d2778420b0 INFO     │  ┝━ ｉ [info]: denied ❌ - no entries were released | event_tag_id: 11
kanidmd  | 76071ae1-1f1b-40b6-a9f8-d1d2778420b0 ERROR    │  ┝━ 🚨 [error]: modify: no candidates match filter, failure Filter(Valid) (not ( (class eq Iutf8("tombstone") or class eq Iutf8("recycled")) ) and (uuid eq Uuid(fc4a6f61-a843-4dbf-bf27-cfc91b0ec0f6) and user_auth_token_session eq Refer(d8ebf915-0b57-4439-b3d9-ea757cf7585e)))
kanidmd  |  | event_tag_id: 4
kanidmd  | 76071ae1-1f1b-40b6-a9f8-d1d2778420b0 ERROR    │  ┕━ 🚨 [error]: Failed to destroy user auth token NoMatchingEntries | event_tag_id: 1
kanidmd  | 76071ae1-1f1b-40b6-a9f8-d1d2778420b0 WARN     ┕━ 🚧 [warn]:  | latency: 4.545756ms | status_code: 404 | kopid: "76071ae1-1f1b-40b6-a9f8-d1d2778420b0" | msg: "client error"`
![kanidm-signout](https://github.com/kanidm/kanidm/assets/681680/b3bcabbc-d99e-4d75-824b-24b8fe6be996)

What Am I doing wrong ? :)

**Comment by MrSpock** [comment]
Additionaly it looks like there is some invisible overlay on the webpage since its darker and I can't clitk "Return to login"

**Comment by Firstyear** [comment]
If you logout *rapidly* after you login the session may not have propagated to the database or via replication yet. This is because when you login we asynchronously write the session back to the account. So if you login then logout immediately, you "race" the async writeback, and if you beat it, you get that error above.

If you have replication a similar race condition can occur. 

I wonder if in this case since the auth token is within the grace window, and since it's being used to logout, we should just write a session stub with the revoked state which will cause the async write back to be dropped? It would resolve this error.
